Formatted/Added docstrings to all modules in pyvcloud to adhere to sphinx guidelines.

rocknes opened the pull request:
* Fixed sphinx doc generation warnings
* Formatted/Added docstrings for most of the methods of all pyvcloud modules.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/pyvcloud/253)
<!-- Reviewable:end -->

hodgesrm commented:
This is a really impressive clean-up to documentation.  I found only a couple of typos.  

I would approve it immediately but it seems there may be a problem with rebuilding.  I cloned your repo and checked out  the docs branch.  I ran the following commands to build: 'sphinx-apidoc -f --separate -o source ../pyvcloud; make html`.  On looking at the HTML in a browser all the modules were gone. :(

On inspection it appears that the sphinx-apidoc command is wiping out sub-module contents in pyvcloud.rst on rebuild.  This may be the issue. 

---

Reviewed 26 of 26 files at r1.
Review status: all files reviewed, 5 unresolved discussions (waiting on @rocknes)

---

*[pyvcloud/system_test_framework/environment.py, line 91 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExQzO10aO0zXD0mA9p:-LExQzO10aO0zXD0mA9q:b-79eq1k) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/system_test_framework/environment.py#L91)):*
> ```Python
>             file.
> 
>         :return: Nothing
> ```

Is this a general convention?  Seems as though we should just omit if there's no return. Python functions actually seem to return None. (This is equivalent omitting a return statement if you test in the repl.)

---

*[pyvcloud/vcd/acl.py, line 31 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExWrxaC-sriwaFsAYK:-LExWrxaC-sriwaFsAYL:b-x5b9p3) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/acl.py#L31)):*
> ```Python
> 
>         :param pyvcloud.vcd.client.Client client: the client that will be used
>             to make ReST calls to vCD.
> ```

ReST should be REST.  (I think I have seen this variant but it's unusual.)

---

*[pyvcloud/vcd/utils.py, line 505 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExaeJW2ZR4XshE9Hzt:-LExaeJW2ZR4XshE9Hzu:b-489i3z) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/utils.py#L505)):*
> ```Python
> 
> def filter_attributes(resource_type):
>     """Returns a list of arrtibutes for a given resource type.
> ```

typo: arrtributes should be attributes

---

*[pyvcloud/vcd/utils.py, line 507 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExam0x02zEfl9XOCw_:-LExam0x02zEfl9XOCwa:b-1fxz5h) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/utils.py#L507)):*
> ```Python
>     """Returns a list of arrtibutes for a given resource type.
> 
>     :param str resrouce_type: type of resource whose list of attributes we want
> ```

typo: resrouce_type should be resource_type

---

*[pyvcloud/vcd/vapp.py, line 96 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExb4tJ-An2VGd6w0YZ:-LExb4tKA-N68tYVPSBr:b-a25bcg) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/vapp.py#L96)):*
> ```Python
>         :rtype: str
> 
>         :raises: Exception: if the named vm or it's NIC information can't be
> ```

Typo: it's should be its.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/253#-:-LExcsqv5eZ7LLblwbx5:b-1rnjap)*
<!-- Sent from Reviewable.io -->

hodgesrm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  

---

Reviewed 15 of 15 files at r2.
Review status: all files reviewed, 5 unresolved discussions (waiting on @rocknes)

---



*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/253#-:-LEzmcW6-yZtbI4Z_UPN:b593w5t)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: all files reviewed, 5 unresolved discussions (waiting on @rocknes)

---

*[pyvcloud/system_test_framework/environment.py, line 91 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExQzO10aO0zXD0mA9p:-LEzIFEs0FRr3Oj3M1MU:b-kvu3uh) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/system_test_framework/environment.py#L91)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Is this a general convention?  Seems as though we should just omit if there's no return. Python functions actually seem to return None. (This is equivalent omitting a return statement if you test in the repl.)
</blockquote></details>

Sounds good. Removed all ':return: Nothing' lines.

---

*[pyvcloud/vcd/acl.py, line 31 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExWrxaC-sriwaFsAYK:-LEzL7Pm2DHRU4GzRP1D:bqjcj29) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/acl.py#L31)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

ReST should be REST.  (I think I have seen this variant but it's unusual.)
</blockquote></details>

Ankit used to write it that way in our opneapi days :P I picked it up from him. Changing to REST.

---

*[pyvcloud/vcd/utils.py, line 505 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExaeJW2ZR4XshE9Hzt:-LEzLa1t4ItiXp8SOZo-:b-886gsx) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/utils.py#L505)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

typo: arrtributes should be attributes
</blockquote></details>

Fixed

---

*[pyvcloud/vcd/utils.py, line 507 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExam0x02zEfl9XOCw_:-LEzLZ31EXoD_0E9ZzlE:b-886gsx) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/utils.py#L507)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

typo: resrouce_type should be resource_type
</blockquote></details>

Fixed

---

*[pyvcloud/vcd/vapp.py, line 96 at r1](https://reviewable.io/reviews/vmware/pyvcloud/253#-LExb4tJ-An2VGd6w0YZ:-LEzLvYKEf64cae8Ny2G:b4gi8uw) ([raw file](https://github.com/vmware/pyvcloud/blob/49f81e120bacb8e453d63114706ed4191741c92e/pyvcloud/vcd/vapp.py#L96)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Typo: it's should be its.
</blockquote></details>

Fixed. Fund few more instances of it's :P fixed those too.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/253)*
<!-- Sent from Reviewable.io -->
